examples/hello_rust/hello_rust_main.rs

```diff
@@ -21,54 +21,115 @@
 /****************************************************************************
  * Attributes
  ****************************************************************************/
+ 
+ #![no_main]
+ #![no_std]
+ 
+ /****************************************************************************
+  * Uses
+  ****************************************************************************/
+ 
+ use core::panic::PanicInfo;
+ use core::ffi::{ c_char, c_int, c_void };
+ 
+ /****************************************************************************
+  * Externs
+  ****************************************************************************/
+ 
+ extern "C" {
+     pub fn printf(format: *const u8, ...) -> i32;
+     pub fn puts(s: *const c_char) -> c_int;
+     pub fn fgets(
+         buf: *mut c_char,
+         n: c_int,
+         stream: *mut c_void
+     ) -> *mut c_char;
+     pub fn lib_get_stream(fd: c_int) -> *mut c_void;
+ 
+     // Functions for Rust LED Blinky example
+     pub fn open(path: *const u8, oflag: i32, ...) -> i32;
+     pub fn read(fd: i32, buf: *mut u8, count: u32) -> i32;
+     pub fn write(fd: i32, buf: *const u8, count: u32) -> i32;
+     pub fn close(fd: i32) -> i32;
+     pub fn ioctl(fd: i32, request: i32, ...) -> i32;  //  On NuttX: request is i32, not u64 like Linux
+     pub fn sleep(secs: u32) -> u32;
+     pub fn usleep(usec: u32) -> u32;
+     pub fn exit(status: u32) -> !;
+ }
+ 
+ /****************************************************************************
+  * Rust Constants
+  ****************************************************************************/
+ pub const O_WRONLY: i32 = 1 << 1;
+ pub const ULEDIOC_SETALL: u16 = 0x1d03;
+ 
+ /****************************************************************************
+  * Private functions
+  ****************************************************************************/
+ 
+ /****************************************************************************
+  * Panic handler (needed for [no_std] compilation)
+  ****************************************************************************/
+ 
+ #[panic_handler]
+ fn panic(_panic: &PanicInfo<'_>) -> ! {
+     loop {}
+ }
+ 
+ /****************************************************************************
+  * Public functions
+  ****************************************************************************/
+ 
+ /****************************************************************************
+  * hello_rust_main
+  ****************************************************************************/
+ 
+ #[no_mangle]
+ pub extern "C" fn main(_argc: i32, _argv: *const *const u8) -> i32 {
```

Comment:
```suggestion
 pub extern "C" fn hello_rust_main(_argc: i32, _argv: *const *const u8) -> i32 {
```

Let's restore `main` to `hello_rust_main`. Remember that this fix is only needed for Ox64 and NuttX Kernel Mode. QEMU won't need this.

Reply:
Okay.